src/main/java/ui/components/pickers/PickerLabel.java

```diff
@@ -1,47 +1,41 @@
 package ui.components.pickers;
 
 import backend.resource.TurboLabel;
+
 import com.sun.javafx.tk.FontLoader;
 import com.sun.javafx.tk.Toolkit;
+
 import javafx.scene.Node;
 import javafx.scene.control.Label;
 import javafx.scene.control.Tooltip;
 
-// for use with LabelPickerDialog
+/**
+ * This class is to represent a label in LabelPickerDialog
+ *
+ * It contains attributes such as selected, highlighted, removed and faded in order
+ * to produce the appropriate styled node through getNode()
+ */
 public class PickerLabel extends TurboLabel {
 
-    private final LabelPickerUILogic labelPickerUILogic;
     private boolean isSelected;
     private boolean isHighlighted;
     private boolean isRemoved;
     private boolean isFaded;
-    private final boolean isTop;
+    private final boolean canDisplayFullName;
 
-    public PickerLabel(TurboLabel label, LabelPickerUILogic labelPickerUILogic, boolean isTop) {
-        super(label.getRepoId(), label.getColour(), label.getActualName());
-        this.labelPickerUILogic = labelPickerUILogic;
+    public PickerLabel(TurboLabel label, boolean canDisplayFullName) {
+        super(label.getRepoId(), label.getColour(), label.getFullName());
         isSelected = false;
         isHighlighted = false;
         isRemoved = false;
         isFaded = false;
-        this.isTop = isTop;
-    }
-
-    public PickerLabel(TurboLabel label, LabelPickerUILogic labelPickerUILogic,
-                       boolean isSelected, boolean isHighlighted, boolean isRemoved, boolean isFaded, boolean isTop) {
-        super(label.getRepoId(), label.getColour(), label.getActualName());
-        this.labelPickerUILogic = labelPickerUILogic;
-        this.isSelected = isSelected;
-        this.isHighlighted = isHighlighted;
-        this.isRemoved = isRemoved;
-        this.isFaded = isFaded;
-        this.isTop = isTop;
+        this.canDisplayFullName = canDisplayFullName;
     }
 
     @Override
     public Node getNode() {
         // actual name for labels at the top, add tick for selected labels
-        Label label = new Label((isTop ? getActualName() : getName()));
+        Label label = new Label((canDisplayFullName ? getFullName() : getShortName()));
         label.getStyleClass().add("labels");
         if (isRemoved) label.getStyleClass().add("labels-removed"); // add strikethrough
         String style = getStyle() + (isHighlighted ? " -fx-border-color: black;" : ""); // add highlight border
```

Comment:
Should this be split into 2 lines? Since the line after this is simply adding to the style.
```
String style = getStyle();
style = style += ... // add highlight border
stye = style += ... // change opacity if needed.
```